src/dvc_data/objects/tree.py

```diff
@@ -64,7 +64,9 @@ def _trie(self):
     def add(
         self, key: Tuple[str, ...], meta: Optional["Meta"], oid: "HashInfo"
     ):
-        self.__dict__.pop("trie", None)
+        trie = self.__dict__.get("_trie")
+        if trie is not None:
+            trie.update([(key, (meta, oid))])
         self._dict[key] = (meta, oid)
```

Comment:
The idea is that we won't waste time while building the tree (building trie might be time consuming). So I would worry a bit about this affecting `build()` time.